knotx-knot/knotx-knot-handlebars/src/main/java/io/knotx/knot/templating/impl/HandlebarsKnotProxyImpl.java

```diff
@@ -57,10 +82,8 @@ public HandlebarsKnotProxyImpl(HandlebarsKnotConfiguration configuration) {
         knotContext.setTransition(DEFAULT_TRANSITION);
         Optional.ofNullable(knotContext.getFragments()).ifPresent(fragments ->
             fragments.stream()
-                .filter(fragment -> fragment.knots().contains(SUPPORTED_FRAGMENT_KNOT))
-                .forEach(fragment -> fragment.content(startComment() +
-                    new HandlebarsFragment(fragment).compileWith(handlebars)
-                    + endComment()))
+                .filter(fragment -> shouldProcess(Sets.newHashSet(fragment.knots())))
+                .forEach(fragment -> fragment.content(evaluate(template(fragment), fragment.context())))
```

Comment:
two many methods in methods, make it more readable, e.g.:
```
fragment.content(evaluate(fragment));
```
and in the evaluate() method just execute template() method and fragment.context()

Reply:
done